posthog/src/main/java/com/posthog/internal/PostHogQueue.kt

```diff
@@ -163,32 +174,33 @@ internal class PostHogQueue(
 
         val events = mutableListOf<PostHogEvent>()
         for (file in files) {
-            try {
-                val inputStream = config.encryption?.decrypt(file.inputStream()) ?: file.inputStream()
-                inputStream.use {
-                    val event = config.serializer.deserialize<PostHogEvent?>(it.reader().buffered())
-                    event?.let { theEvent ->
-                        events.add(theEvent)
+            if (file.isStreamable()) {
+                try {
+                    events.add(file.event(config))
+                } catch (e: Throwable) {
+                    synchronized(dequeLock) {
+                        deque.remove(file)
                     }
+                    file.deleteSafely(config)
+                    config.logger.log("File: ${file.name()} failed to parse: $e.")
                 }
-            } catch (e: Throwable) {
-                synchronized(dequeLock) {
-                    deque.remove(file)
-                }
-                file.deleteSafely(config)
-                config.logger.log("File: ${file.name} failed to parse: $e.")
+            } else {
+                events.add(file.event(config))
             }
         }
 
         var deleteFiles = true
         try {
+            // TODO: python checks if the payload isnt bigger than 900kb
```

Comment:
The tricky part here is that we serialize/stream from the disk directly to the network because of performance reasons (android specifically), so we don't load everything in memory.
if we wanna check the size before I'd need to change that and have one implementation per platform or pay the performance price on android.
Since we don't have this check yet on Android either (events are small anyways), I will keep as like this and implement whenever is needed

Reply:
> events are small anyways

...wellllll until someone puts something massive in a property 🙈 	

posthog-js truncates properties to a configurable max length (and doesn't enforce a max number of properties) but doesn't otherwise check the size so it must be ok to let capture fail here...

Reply:
https://github.com/PostHog/posthog-android/pull/130 at least reduces the batch size on the next retry so it's only gonna drop the event entirely if a single event is bigger than our limits